File: src/o1tkeklargestnumber/TS/KthLargest.js
"use strict";
// File: KthLargest.ts
Object.defineProperty(exports, "__esModule", { value: true });
exports.KthLargest = void 0;
const MinHeap_1 = require("./MinHeap");
class KthLargest {
    constructor(k, nums) {
        this.k = k;
        this.topKHeap = new MinHeap_1.MinHeap();
        nums.forEach(num => this.add(num));
    }
    add(val) {
        if (this.topKHeap.length() < this.k) {
            this.topKHeap.push(val);
        }
        else if (val > (this.topKHeap.peek() || -Infinity)) {
            this.topKHeap.pop();
            this.topKHeap.push(val);
        }
        return this.topKHeap.peek() || -1;
    }
}
exports.KthLargest = KthLargest;

File: src/o1tkeklargestnumber/Python/kth_largest.py
# File: kth_largest.py

from minheap import MinHeap


class KthLargest:
    def __init__(self, k, nums):
        self.k = k
        self.min_heap = MinHeap()
        for num in nums:
            self.add(num)

    def add(self, val):
        if self.min_heap.size() < self.k:
            self.min_heap.push(val)
        elif val > self.min_heap.top():
            self.min_heap.pop()
            self.min_heap.push(val)
        return self.min_heap.top()
